vpr/src/draw/draw.cpp

```diff
@@ -976,16 +980,56 @@ static void drawplace(ezgl::renderer* g) {
                 //Determine the block color
                 ezgl::color block_color;
                 t_logical_block_type_ptr logical_block_type = nullptr;
+#ifdef  VTR_ENABLE_DEBUG_LOGGING
+                if(f_placer_debug){ 
+                    t_pl_loc curr_loc;
+                    curr_loc.x = i;
+                    curr_loc.y = j;
+                    auto it = std::find_if(draw_state->colored_blocks.begin(),draw_state->colored_blocks.end(),[&curr_loc](const std::pair<t_pl_loc, ezgl::color>& a){return (a.first.x==curr_loc.x && a.first.y==curr_loc.y);});
```

Comment:
This is a bit awkward, and requires a linear search through the colored block.

Perhaps we should re-consider how this is structured.

Why not store a color for *all* blocks
```cpp
vtr::vector<ClusteredBlockId,ezgl::color> colored_blocks; //Color of each block in the netlist
```
And use a sentinel color (e.g. perhaps define ``ezgl::color DEFAULT_BLOCK_COLOR = ezgl::WHITE``) to mark blocks which aren't highlighted.

For instance the setup code would then be:
```cpp
//All blocks initially use default color (which will be determined by the block-type)
colored_blocks = vtr::vector<ClusteredBlockId,ezgl::color> (clb_nlist.blocks().size(), DEFAULT_BLOCK_COLOR);
```

And the check here would become:
```cpp
if (f_placer_debug && draw_state->colored_blocks[bnum] != DEFAULT_BLOCK_COLOR) {
    block_color = draw_state->colored_blocks[bnum];
} else { //Draw with default block type based-color
    block_color = get_block_type_color(device_ctx.grid[i][j].type);
    block_color = lighten_color(block_color, EMPTY_BLOCK_LIGHTEN_FACTOR);
}
//Draw the block in the specified color
```
which would be much simpler and also more flexible (other code, such as the placer, only needs to mark the block colors).